File: usecases/src/test/java/com/zeyad/usecases/integration/Success.java
package com.zeyad.usecases.integration;

/**
 * @author by ZIaDo on 7/7/17.
 */
public class Success {

    private boolean success;

    public Success(boolean b) {
        success = b;
    }

    @Override
    public int hashCode() {
        return (success ? 1 : 0);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        Success success1 = (Success) o;
        return success == success1.success;
    }
}

File: usecases/src/test/java/com/zeyad/usecases/integration/Success.kt
package com.zeyad.usecases.integration

/**
 * @author by ZIaDo on 7/7/17.
 */
class Success(private val success: Boolean) {

    override fun hashCode(): Int {
        return if (success) 1 else 0
    }

    override fun equals(o: Any?): Boolean {
        if (this === o)
            return true
        if (o == null || javaClass != o.javaClass)
            return false
        val success1 = o as Success?
        return success == success1!!.success
    }
}
